### PR TITLE
Update doc structure, lint and link check, add placeholder for "onboarding guide"

### PR DESCRIPTION
**Pull request by jfomhover**
## Purpose

* Change documentation structure to highlights quickstarts / tutorials / concepts structure.
* Add placeholder for onboarding guide
* Apply linting rules to all pages
* Verify internal links between pages

## What is the expected review turnaround time?

Urgency:
- [x] Medium (needs review within a few days - most common)

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,25 +6,30 @@ This repo provides some code samples for running a federated learning pipeline i
...
-- [Support](#need-support)
+- [Need Support?](#need-support)
 - [Contributing](#contributing)
 
 ### :zap: Getting Started
 
-No time to read? Get directly to the [**quickstart**](./docs/quickstart.md) to provision a demo within minutes in your own subscription.
+No time to read? Go directly to the [**quickstart**](./docs/quickstart.md) to provision a demo within minutes in your own subscription.
+
+To understand the journey to acheve production-ready Federated Learning, please check our guide to [**Plan Your FL Project**](./docs/concepts/plan-your-fl-project.md).
```
Comment (garg-amit) [inline README.md:22]: typo `achieve`

File: docs/README.md [hunk 2]
```diff
@@ -2,59 +2,44 @@
...
-| Folder | Description |
-| :--- | :--- |
-| [examples](../examples) | Scripts and pipelines to run FL sample experiments. |
-| [mlops](../mlops) | Provisioning scripts. See instructions [here](./provisioning/README.md). |
+## Quickstart
 
+No time to read? Get directly to the [**quickstart**](./quickstart.md) to provision a demo within minutes in your own subscription.
 
-## Tutorial on how to adapt the "literal" and the "factory" code
+To understand the journey to acheve production-ready Federated Learning, please check our guide to [**Plan Your FL Project**](./concepts/plan-your-fl-project.md).
```
Comment (garg-amit) [inline docs/README.md:40]: typo `achieve`

**Review by garg-amit — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (jfomhover) [reply README.md]: good catch thx!

File: docs/README.md (hunk 2 above)
Comment (jfomhover) [reply docs/README.md]: good catch thx!

**Pull request merged**